builders/interoperability/xcm/xc-registration/assets.md

```diff
@@ -0,0 +1,434 @@
+
+You can use the [`xcm-asset-registrator.ts` script](https://github.com/Moonsong-Labs/xcm-tools/blob/main/scripts/xcm-asset-registrator.ts){target=_blank} to calculate the encoded calldata and even submit the preimage and proposal if you desire. If you're registering an asset and opening a channel, you'll want to wait to submit the preimage and proposal until you have the calldata for the channel-related calls.
+
+To get the encoded calldata for the `assetManager.registerForeignAsset` extrinsic, you can use the following arguments:
+
+- `--ws-provider` or `--w` - the WebSocket provider to be used for the requests. The WSS network endpoints for each Moonbeam-based network are as follows:
+
+    === "Moonbeam"
+
+        ```text
+        wss://wss.api.moonbeam.network
+        ```
+
+    === "Moonriver"
+
+        ```text
+        wss://wss.api.moonriver.moonbeam.network
+        ```
+
+    === "Moonbase Alpha"
+
+        ```text
+        {{ networks.moonbase.wss_url }}
+        ```
+
+- `--asset` or `--a` - the multilocation of the asset
+- `--name` or `--n` - the name of the asset
+- `--symbol` or `--sym` - the symbol of the asset. **Remember that "xc" should be prepended to the symbol to indicate the asset is an XCM-enabled asset**
+- `--decimals` or `--d` - the number of decimals of the asset
+- `--existential-deposit` or `--ed` - (optional) - the existential deposit of the registered asset. This should always be set to `1`
+- `--sufficient` or `--suf` - (optional) - the sufficiency, which dictates whether an asset can be sent to an account without a native token balance. This should always be set to `true`
+
+To create a batch transaction that also sets the units per second or revert code of the asset's precompile in addition to the asset registration, you can choose to add these arguments:
+
+- `--units-per-second` or `--u` - (optional) - the units per second, which specifies the amount to charge per second of execution in the registered asset. You should have calculated this value in the [previous section](#calculate-units-per-second). If this is provided, the script will create a batch transaction for the governance proposal that, at a minimum, will register the asset and set the units per second on-chain
+- `--revert-code` or `--revert` - (optional) - registers the revert code for the asset's precompile in the EVM. If this is provided, the script will create a batch transaction for the governance proposal that, at a minimum, will register the asset and set the revert code
+
+    !!! remember "Warning"
+        For chains with [OpenGov](/learn/features/governance#opengov){target=_blank}, the `--revert-code true` flag was removed. This flag includes a `system.setStorage` call that the General Admin Origin can't execute. The dummy EVM bytecode can be set later with a call to the Precompile Registry precompile, which means that you don't need to worry about going through governance to set the revert code!
```

Comment:
If this gets merged before RT240x gets rolled out to Moonbeam, I'll need to adjust this. Otherwise, I can update it in https://github.com/moonbeam-foundation/moonbeam-docs/pull/694

Reply:
Go for it assuming Moonbeam has OpenGov already. Thera re no new XCM Integrations being worked on right now